staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,6 +181,20 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// The TransformFunc is called for each object which is about to be stored.
+	//
+	// This function is intended for you to take the opportunity to
```

Comment:
It would be helpful to list the limitations of this. Does the returned type have to be the same type as the input?

From what I can tell it does not but must implement `runtime.Object` and `metav1.ObjectMetaAccessor` or equivalent interfaces